testing/internal/e2e/helpers.go

```diff
@@ -0,0 +1,53 @@
+package e2e
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"os"
+	"os/exec"
+	"testing"
+)
+
+// CommandResult encapsulates the output running an external command
+type CommandResult struct {
+	Stdout   []byte
+	Stderr   []byte
+	ExitCode int
+}
+
+const envToCheck = "E2E_PASSWORD_AUTH_METHOD_ID"
+
+// RunCommand executes external commands on the system. Returns the results
+// of running the provided command.
+func RunCommand(name string, args ...string) (*CommandResult, error) {
+	var outbuf, errbuf bytes.Buffer
+
+	cmd := exec.Command(name, args...)
+	cmd.Stdout = &outbuf
+	cmd.Stderr = &errbuf
+
+	err := cmd.Run()
+
+	var ee *exec.ExitError
+	var exitCode int
+	if errors.As(err, &ee) {
+		exitCode = ee.ExitCode()
+	}
+
+	return &CommandResult{
+		Stdout:   outbuf.Bytes(),
+		Stderr:   errbuf.Bytes(),
+		ExitCode: exitCode,
+	}, err
```

Comment:
It is unusual to return an error and a result at the same time. Could we add a comment that the returned `*CommandResult` is always valid, even if there is an error, to the function comment?